state/upgrades_test.go

```diff
@@ -4995,6 +4995,229 @@ func (s *upgradesSuite) TestTranslateK8sServiceTypes(c *gc.C) {
 	)
 }
 
+func (s *upgradesSuite) TestUpdateLegacyKubernetesCloudCredentialsCertificate(c *gc.C) {
+	cloudColl, cloudCloser := s.state.db().GetRawCollection(cloudsC)
+	defer cloudCloser()
+	cloudCredColl, cloudCredCloser := s.state.db().GetRawCollection(cloudCredentialsC)
+	defer cloudCredCloser()
+
+	_, err := cloudColl.RemoveAll(nil)
+	c.Assert(err, jc.ErrorIsNil)
+	_, err = cloudCredColl.RemoveAll(nil)
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = cloudColl.Insert(bson.M{
+		"_id":        "localhost",
+		"name":       "localhost",
+		"type":       "kubernetes",
+		"auth-types": []string{"empty"},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = cloudCredColl.Insert(bson.M{
+		"_id":       "localhost#admin#streetcred",
+		"owner":     "admin",
+		"cloud":     "localhost",
+		"name":      "streetcred",
+		"revoked":   false,
+		"auth-type": "certificate",
+		"attributes": map[string]string{
+			"Token":   "this-is-my-token",
+			"rbac-id": "rbac",
+		},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	expectedClouds := []bson.M{{
+		"_id":  "localhost",
+		"name": "localhost",
+		"type": "kubernetes",
+		"auth-types": []interface{}{
+			"certificate",
+			"clientcertificate",
+			"oauth2",
+			"oauth2withcert",
+			"userpass",
```

Comment:
I don't think we want to include the depecated auth types after upgrade - we need ti support reading them in from an existing yaml file etc but we should not write them out here

Reply:
Done & Done